mint-client/src/clients/user.rs

```diff
@@ -323,6 +326,65 @@ impl UserClient {
             .await
             .map_err(|_| ClientError::WaitContractTimeout)?
     }
+
+    pub async fn create_invoice_and_offer<R: RngCore + CryptoRng>(
+        &self,
+        amount: Amount,
+        mut rng: R,
+    ) -> Result<(KeyPair, Invoice), ClientError> {
+        let (payment_keypair, payment_public_key) =
+            self.context.secp.generate_schnorrsig_keypair(&mut rng);
+        let raw_payment_secret = payment_public_key.serialize();
+        let payment_hash = bitcoin::secp256k1::hashes::sha256::Hash::hash(&raw_payment_secret);
+        let payment_secret = PaymentSecret(raw_payment_secret);
+
+        // Temporary lightning node pubkey
+        let (node_secret_key, node_public_key) = self.context.secp.generate_keypair(&mut rng);
+
+        let invoice = InvoiceBuilder::new(Currency::Regtest)
+            .amount_milli_satoshis(amount.milli_sat)
+            .description("".into())
+            .payment_hash(payment_hash)
+            .payment_secret(payment_secret)
+            .current_timestamp()
+            .min_final_cltv_expiry(144)
```

Comment:
Where does this number come from?

Reply:
Ahh I just copied that from an example. I’m not sure what we should put in for all those values …

Reply:
This parameter "is the minimum difference between HTLC CLTV timeout and the current block height" (https://github.com/lightning/bolts/blob/master/02-peer-protocol.md#cltv_expiry_delta-selection).

Eric observed that "consensus block height" for the federation trails the real bitcoin network because of the pegging in/out stuff, but the payer of the invoice actually applies this difference so that's not relevant here.

Updated to use core lightning default 
![image](https://user-images.githubusercontent.com/4335621/174396061-5c77cc43-7413-4c58-9276-2e0c85a9f58b.png)

Reply:
At some point we should discuss the details with someone who has a deeper understanding of LN (optimizations, security relevant corner cases), for now I think it should work.